sdk/attestation/Azure.Security.Attestation/src/Models/AttestationSigner.cs

```diff
@@ -19,19 +20,24 @@ public class AttestationSigner
         /// </summary>
         /// <param name="signingCertificates"></param>
         /// <param name="certificateKeyId"></param>
-        public AttestationSigner(X509Certificate2[] signingCertificates, string certificateKeyId)
+        public AttestationSigner(IEnumerable<X509Certificate2> signingCertificates, string certificateKeyId)
         {
-            SigningCertificates = signingCertificates;
+            SigningCertificates = signingCertificates switch
+            {
+                IReadOnlyList<X509Certificate2> => (IReadOnlyList<X509Certificate2>)signingCertificates,
```

Comment:
```suggestion
                IReadOnlyList<X509Certificate2> certificateList => certificateList
```

Justification:  The `is` check has to happen for pattern matching, by not capturing the result, you're paying the cost twice due to the redundant cast.